Glue templates

ibrand opened the pull request:
Glue job templating done such that the glue job pulls the latest signatures_job.py script from csl when we terraform deploy so we don't need to manually update the glue code anymore

ibrand committed:
glue template update to pull directly from the csl python script for populating the glue job

ibrand committed:
glue terraform variable additions to the locals block

ibrand committed:
copy in another locals block from the csl version

ibrand committed:
go back to the data syntax

ibrand committed:
try fully replacing our glue file with csl's

sjwmoveon commented on glue_job.tf:
```diff
@@ -4,20 +4,14 @@ resource "aws_glue_catalog_database" "catalog_db" {
   database_name = aws_glue_catalog_database.catalog_db.name
   name = "${var.controlshift_environment}_full_signatures"
   role = aws_iam_role.glue_service_role.arn
-  configuration = jsonencode(
```
Why are we getting rid of this bit?

sjwmoveon replied:
https://github.com/MoveOnOrg/terraform-aws-controlshift-redshift-sync/commit/4a2e1c32a1036e892a3e6fb647c2b7d51fd34b9c looks like we manually set this part of the configuration so we may want to keep it.

sjwmoveon commented on glue_job.tf:
```diff
@@ -26,45 +20,60 @@ resource "aws_glue_crawler" "signatures_crawler" {
+  acl    = "private"
+}
 
-  lifecycle_rule {
```
Seems like we would want to keep this rule for deleting temp files?

sjwmoveon replied:
Similarly we added this manually added lifecycle rule to the config: https://github.com/MoveOnOrg/terraform-aws-controlshift-redshift-sync/commit/19be449c3420f05b3f590e2baa8611d59472ae67 and https://github.com/MoveOnOrg/terraform-aws-controlshift-redshift-sync/commit/7fa1de1bf76bd0739438572e8cfcea5c1b6b4113

sjwmoveon reviewed (changes requested):
This mostly makes sense but I have a couple questions.

ibrand commented:
So, this was a direct copypaste from https://github.com/controlshift/terraform-aws-controlshift-redshift-sync/blob/main/glue_job.tf which is why some of those blocks were removed. I think either one or both of those blocks might need to be written differently in the newer terraform format (for example pasting the `lifecycle_rule` block back in throws errors) -- do you think it's worth debugging?

sjwmoveon commented:
I think it's worth debugging since both changes were made intentionally on our end to do some extra cleanup.

sjwmoveon committed:
Use `character varying` as output type for Redshift

In Redshift, `string` is not a supported column type.  Replace `string` with `character varying` for output columns.

sjwmoveon commented on glue_job.tf:
```diff
@@ -153,8 +162,6 @@ resource "aws_glue_job" "signatures_full" {
   name = "cs-${var.controlshift_environment}-signatures-full"
   connections = [ aws_glue_connection.redshift_connection.name ]
   glue_version = "3.0"
-  number_of_workers = 9
```
This also was set manually

sjwmoveon committed:
Add back additional manual changes

sjwmoveon committed:
Remove extra expiration block

ibrand commented:
Oh I see that you added the manual changes back in! Does this mean that you got it to deploy without errors with them?

sjwmoveon commented:
I did, yes!  No luck with the results of the Glue job, though.

ibrand commented:
I'd say we're good to merge in then!

sjwmoveon reviewed (approved):
Approved with the more recent changes.  @ibrand you may want to take another look as well since I made the last few commits.

sjwmoveon reviewed (changes requested):
I think we want to undo all of this since we're not going with the templated Glue job any more.  Fortunately it was never checked in.

ibrand commented:
Great! I didn't realize it was never checked in. Closing this.
